src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaseEquivalences.cs

```diff
@@ -0,0 +1,167 @@
+        ///                                  EquivalenceCasingMap[index2] is 0xFFFF then <paramref name="c"/> isn't involved in case conversion so we return false. Otherwise,
+        ///                                  we decompose the ushort into the highest 3 bits, and the other 13 to compute two different numbers: the number of equivalence characters
+        ///                                  to grab from the third table (highest 3 bits and save it as count), and the index (aka. index3) to grab them from (other 13 bits).
+        ///       EquivalenceCasingValues => The final table contains ushort representing characters. We grab the index3 computed in the previous table, and we use it
+        ///                                  to search on this table and grab the next 'count' items which are the equivalence mappings for <paramref name="c"/>.
+        ///
+        /// Example: using character 'A' (0x0041). We caluclate index by doing `index = 0x0041 / 1024` which results in 0. We then look on the first lookup table using the
+        /// calculated index `EquivalenceFirstLevelLookup[index]` which results in the value 0x0000. Because this value is not 0xFFFF, character 'A' may be participating in case
+        /// conversion, so we continue our search by looking into the second lookup table. We calculate index2 by doing `index2 = (0x0041 % 1024) + 0x0000` which results in
+        /// index2 = 0x0041. We then use that index to search in the second lookup table `EquivalenceCasingMap[0x0041]` and get a value of 0x4000 back. Because that value isn't
+        /// 0xFFFF then we now know that the character 'A' participates in case conversion. We decompose the value we got from the second table 0x4000 (0b_0100_0000_0000_0000 in binary)
+        /// into the 3 highest bits (0b_010) and the rest of the 13 bits (0b_0_0000_0000_0000) resulting in: count = 2 and index3 = 0. This means that we finally must go into the
+        /// third lookup table EquivalenceFirstLevelLookup at index 0, and grab the next 2 items which gives us 0x0041 ('A') and 0x0061 ('a'), which are the two characters considered
+        /// equivalent in a case comparison for 'A'.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TryFindCaseEquivalencesForChar(char c, out ReadOnlySpan<char> equivalences)
+        {
+            equivalences = null;
+
+            // Right shifting by 10 in order to divide by 1024 (CharactersPerRange)
+            Debug.Assert((c >> 10) < 0xFF);
+            byte index = (byte)(c >> 10);
+            ushort FirstLevelLookupValue = EquivalenceFirstLevelLookup[index];
+
+            // If character belongs to a range that doesn't participate in casing, then just return false
+            if (FirstLevelLookupValue == 0xFFFF)
+            {
+                return false;
+            }
+
+            equivalences = PerformSecondLevelLookup();
+            return equivalences != null;
```

Comment:
As noted elsewhere, comparing span against null is generally not what you want to do.